### PR TITLE
Travis CI: add clang-format check

### PR DESCRIPTION
**Pull request by antiagainst**
Fixes https://github.com/KhronosGroup/SPIRV-Tools/issues/1029

### CODE REVIEW

File: utils/check_code_format.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# This script assumes to be invoked at the project root directory.
+
+FILES_TO_CHECK=$(git diff --name-only master | grep -E ".*\.(cpp|cc|c\+\+|cxx|c|h|hpp)$")
+
+if [ -z "${FILES_TO_CHECK}" ]; then
+  echo "No source code to check for formatting."
+  exit 0
+fi
+
+FORMAT_DIFF=$(git diff -U0 master -- ${FILES_TO_CHECK} | ./utils/clang-format-diff.py -p1 -style=Google)
```
Comment (pierremoreau) [inline utils/check_code_format.sh]: Shouldn’t it rather be using the style defined in the .clang-format files (which is the Google style, but with one additional tweak)?

File: utils/clang-format-diff.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+#
+#===- clang-format-diff.py - ClangFormat Diff Reformatter ----*- python -*--===#
+#
+#                     The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
```
Comment (dnovillo) [inline utils/clang-format-diff.py]: LICENSE.TXT should be included here.

File: utils/check_code_format.sh (hunk 1 above)
Comment (antiagainst) [reply utils/check_code_format.sh]: Done

File: utils/clang-format-diff.py (hunk 2 above)
Comment (antiagainst) [reply utils/clang-format-diff.py]: I've changed to download this file for each build.

**Pull request merged**